tests/manual/printing/PrintPageRangeTest.java

```diff
@@ -0,0 +1,153 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javafx.application.Application;
+import javafx.concurrent.Worker;
+import javafx.geometry.Insets;
+import javafx.print.PrinterJob;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+import javafx.stage.Window;
+
+public class PrintPageRangeTest extends Application {
+
+    private static final double NAVI_BAR_MIN_DIMENSION = 32.0;
+    private static final double PADDING_VALUE = 2.0;
+    private static final String buttonStyle = "-fx-font-weight: bold; -fx-font-size: 16px;";
+    private static final String goButtonUnicodeSymbol = "\u21B5";
+    private static final String stopButtonUnicodeSymbol = "\u2715";
+    private static final String reloadButtonUnicodeSymbol = "\u27F3";
+
+    private volatile boolean passed = false;
+    private Scene scene;
+    private WebView webView;
+    private Label bottomMessageLabel;
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+    public void start(Stage stage) {
+        stage.setWidth(640);
+        stage.setHeight(480);
+        stage.setTitle("Printing test with Page Range Option");
+        stage.setScene(createScene());
+        stage.show();
+    }
+
+    static final String instructions =
+                    " 1. Press print button to open Print Dialog.\n" +
+                    " 2. Select a Printer and PageRange 3 to 4 in page range selection.\n" +
+                    " 3. Click Ok.\n" +
+                    " 4. After this 2 pages must be printed.\n" +
+                    " 5. When all pages are printed you will see <END OF PRINT JOB> on bottom.\n" +
+                    " 6. Check whether printed pages are page no 3 and page no 4.\n" +
+                    " 7. if Yes then Test Passed else Test Failed";
+
+    static final String initialURL = "https://en.wikipedia.org/wiki/Java_version_history";
```

Comment:
Have you tested this when using a system proxy ? And do the webview tests in general use random webpages from the internet ? Why can't we generate some simple local content that we KNOW is 4 pages long. You don't even know that for sure about this page.